[18.0][FIX] sale_order_lot_selection/sale_order_product_availability_inline: Don't use demo data for tests / fix display

rousseldenis opened the pull request:
Fixes:

- [x] sale_order_lot_selection: Don't rely on demo data for tests
- [x] sale_order_product_availability_inline: Since Odoo's change here: https://github.com/odoo/odoo/commit/c84b67e5dca1a74e1a5862e754455944015f4105

rousseldenis committed:
[FIX] sale_order_lot_selection: Don't use demo data for tests

rousseldenis commented:
@bosd @astirpe This fixes tests when mrp is installed.

bosd reviewed (approved):
Thanks! Let's get this branch :green_circle: again

bosd commented:
Fix as discovered broken tests in #3667

astirpe commented:
Thank you @rousseldenis !

rousseldenis committed:
[FIX] sale_order_product_availability_online: Display the quantity for templates + fix for variants

Display correctly the quantity even if product template has been displayed instead of variant.

Don't add the quantity if the product name is used in line description
since https://github.com/odoo/odoo/commit/c84b67e5dca1a74e1a5862e754455944015f4105

rousseldenis commented:
@ernestotejeda

rousseldenis commented:
@pedrobaeza As this involves one of yours

rousseldenis commented:
@bosd @astirpe Could you functionaly review `sale_order_product_availability_inline` too on runboat (that one failed too)?

pedrobaeza commented on sale_order_product_availability_inline/models/sale.py:
```diff
@@ -12,3 +12,8 @@ def _compute_name(self):
         return super()._compute_name()
+
+    def _compute_translated_product_name(self):
+        if self.env.context.get("so_product_stock_inline"):
```
Can you add any code comment here explaining why this?

pedrobaeza commented on sale_order_product_availability_inline/views/sale_views.xml:
```diff
@@ -15,6 +15,16 @@
+            >
+                <attribute name="context" operation="update">
+                    {"so_product_stock_inline": True,
+                    "warehouse": parent.warehouse_id}
```
Isn't there here other way for not overriding the context and being fragile if Odoo changes this context upstream (or in new versions)?

rousseldenis replied:
It shouldn't in stable version. `update` is quite safe as they wouldn't remove context at all

rousseldenis replied:
Note: I try here to fix tests in 18.0 branch. Maybe enhancements could be done in a new PR

pedrobaeza replied:
Well, we both know that these things happen in stable, but if you don't have any other way to fix it, OK.

pedrobaeza reviewed (approved):
/ocabot merge patch
